File: clients/python/coyote/models/auth_token_update_in.py
# this file is @generated
import typing as t
from pydantic import Field

from ..internal.base_model import BaseModel


class AuthTokenUpdateIn(BaseModel):
    namespace: t.Optional[str] = None

    id: str

    name: t.Optional[str] = None

    expiry_millis: t.Optional[int] = Field(default=None, alias="expiry_millis")

    metadata: t.Optional[t.Dict[str, str]] = None

    scopes: t.Optional[t.List[str]] = None

    enabled: t.Optional[bool] = None

File: clients/javascript/src/models/authTokenUpdateIn.ts
// this file is @generated

export interface AuthTokenUpdateIn {
    namespace?: string | null;
    id: string;
    name?: string | null;
    expiryMillis?: number | null;
    metadata?: { [key: string]: string } | null;
    scopes?: string[] | null;
    enabled?: boolean | null;
}

export const AuthTokenUpdateInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): AuthTokenUpdateIn {
        return {
            namespace: object['namespace'],
            id: object['id'],
            name: object['name'],
            expiryMillis: object['expiry_millis'],
            metadata: object['metadata'],
            scopes: object['scopes'],
            enabled: object['enabled'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: AuthTokenUpdateIn): any {
        return {
            'namespace': self.namespace,
            'id': self.id,
            'name': self.name,
            'expiry_millis': self.expiryMillis,
            'metadata': self.metadata,
            'scopes': self.scopes,
            'enabled': self.enabled,
        };
    }
}